### PR TITLE
HPCC-27803 fix release symbol issues

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: testing/unittests/CMakeLists.txt [hunk 1]
```diff
@@ -42,8 +44,9 @@ set (    SRCS
          txSummarytests.cpp
          accessmaptests.cpp
          fxpptests.cpp
-         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.cpp;
-         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlAssistant.cpp;
+         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.cpp
+         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlAssistant.cpp
+         ${CMAKE_BINARY_DIR}/generated/ws_loggingservice_esp.cpp
```
Comment (jakesmith) [inline testing/unittests/CMakeLists.txt:49]: @afishbeck - please review + see question re. this and generated export definition in JIRA.

**Comment by ghalliday** [comment]
@afishbeck I am also getting a problem with:

**Comment by ghalliday** [comment]
@afishbeck I am also getting a problem with:

/usr/bin/ld: /home/gavin/dev/hpcc/testing/unittests/loggingtests.cpp:166: undefined reference to `createGetTransactionSeedResponse'

which is missing a DECL_EXPORT (I'm not sure why)

**Comment by afishbeck** [comment]
@ghalliday using those functions externally is rare.  In loggingtests.cpp Tim is calling the service binary interfaces directly.  Normal client code would first create the client interface and use that to make an HTTP call.  So the missing DECL_EXPORT used by extremely rare code like this was probably never noticed.

**Comment by jakesmith** [comment]
> @ghalliday using those functions externally is rare. In loggingtests.cpp Tim is calling the service binary interfaces directly. Normal client code would first create the client interface and use that to make an HTTP call. So the missing DECL_EXPORT used by extremely rare code like this was probably never noticed.



> createGetTransactionSeedResponse

This is the same issue I hit (loggingtests.cpp missing symbol), but different symbol createUpdateLogResponse.
It was fixed by one of the changes in this PR, but adding ${CMAKE_BINARY_DIR}/generated/ws_loggingservice_esp.cpp to the project.
These symbols are in generated files (from .ecm's) , as per question in JIRA,
should all generate factory methods like this be generated with DECL_EXPORT ?
But,

File: testing/unittests/CMakeLists.txt (hunk 1 above)
Comment (afishbeck) [reply testing/unittests/CMakeLists.txt:49]: Regarding question in JIRA: "yes I think those factories should all have the visibility attributes added. Some of those functions are rarely used externally so probably slipped through the cracks. The most common external usage is to go through the client interface. But some of these testing and configuration uses go directly to some of the less common objects."

**Comment by afishbeck** [comment]
As stated inline, yes, I think DECL_EXPORT should be used for all of those factory methods.  Otherwise changes look good.

**Review by afishbeck — APPROVED** [review]

**Comment by afishbeck** [comment]
See https://github.com/hpcc-systems/HPCC-Platform/pull/16244

**Comment by jakesmith** [comment]
@ghalliday - squashed

**Review by richardkchapman — APPROVED** [review]

**Review by ghalliday — APPROVED** [review]

**Pull request merged**